pkg/externalip/api/v1/types.go

```diff
@@ -0,0 +1,28 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ExternalIP maps an external ip address to a service.  The object
+// name is intended to be the ip address so that name uniqueness
+// ensures an ip address is allocated at most once.
+type ExternalIP struct {
```

Comment:
I'd prefer to pick a name which more closely aligns with where we're going to end up.  We've already discussed IngressAddress, or IngressIP.  This is a parallel to an Ingress rule, but for L3.  We'd probably want both services and more complex L4 rules to be able to claim these addresses.  We're starting to use External to mean things off the platform, not ingress to the platform.